armi/bookkeeping/snapshotInterface.py

```diff
@@ -69,7 +69,9 @@ def activateDefaultSnapshots(self):
                 runLog.info(
                     "Adding default snapshot {0} to snapshot queue.".format(snapT)
                 )
+                self.cs.lock = False
```

Comment:
Might be nice to expose the unlock capability through a context manager so that it's more difficult to forget to lock the settings again:
```
with cs.unlock():
    # change stuff
#sally forth
```

Reply:
Yeah, I am not entirely convinced by this `lock` process i have created.  I will do your `unlock()` context manager idea: it's better. 

I guess using the word "lock" here is throwing me for a loop. Because making this a truly thread safe lock would actually be a lot more work than just a little `DeprecationWarning`. Still, since our parallelization is multi-process, not multi-thread, I should probably not worry about it so much.

I'll go do that.

Reply:
Oh yeah, thread safety is way outside the scope.